coderd/database/dbpurge/dbpurge.go

```diff
@@ -30,7 +31,8 @@ func New(ctx context.Context, logger slog.Logger, db database.Store, clk quartz.
 	//nolint:gocritic // The system purges old db records without user input.
 	ctx = dbauthz.AsSystemRestricted(ctx)
 
-	ticker := clk.NewTicker(time.Nanosecond)
+	// Start the ticker with the initial delay.
+	ticker := clk.NewTicker(delay)
```

Comment:
Review: I found this to be the source of a race condition while testing.
I moved the initial 'tick' to instead be inside the goroutine below.

Reply:
What was the race condition?

Reply:
It may actually be the same race condition you mentioned below, just moved around slightly.